File: blitz/F.py


def main():
    s = []
    for _ in range(int(input())):
        cmd, v = list(map(int, input().split()))
        s.append(((lambda x,y: x), ))
        a = True
        if cmd == 1: #max
            while s:
                if s[-1][0] == max and s[-1][1] <= v:
                    s.pop()
                if s[-1][0] == min and s[-1][1] <= v:
                    s.clear()
                    s.append(((lambda x,y: y), v))
                    a = False
                    break
                if s[-1][0] == min and s[-1][1] > v:
                    break
                if s and s[-1][0] == max and s[-1][1] >= v:
                    a = False
                    break
                break
            if a:
                s.append((max, v))
        elif cmd == 2: #min
            while s:
                if s[-1][0] == max and s[-1][1] <= v:
                    break
                if s[-1][0] == min and s[-1][1] <= v:
                    a = False
                    break
                if s[-1][0] == min and s[-1][1] > v:
                    s.pop()
                if s and s[-1][0] == max and s[-1][1] >= v:
                    s.clear()
                    s.append(((lambda x,y: y), v))
                    a = False
                    break
                break
            if a:
                s.append((min, v))

        elif cmd == 3:
            for f, a in s:
                v = f(v,a)
            print(v)

def test_main():
   # Test case 1
   input_values = ['3', '1 10', '2 5', '3']
   expected_output = '10'
   assert main(input_values) == expected_output

   # Test case 2
   input_values = ['4', '2 3', '1 2', '2 1', '3']
   expected_output = '1'
   assert main(input_values) == expected_output

   # Test case 3
   input_values = ['3', '1 10', '1 20', '3']
   expected_output = '20'
   assert main(input_values) == expected_output

   # Test case 4
   input_values = ['3', '2 3', '2 1', '3']
   expected_output = '1'
   assert main(input_values) == expected_output

main()

File: blitz/F.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

type Stack []interface{}

func (s *Stack) IsEmpty() bool {
	return len(*s) == 0
}

func (s *Stack) Push(v interface{}) {
	*s = append(*s, v)
}

func (s *Stack) Pop() interface{} {
	if s.IsEmpty() {
		return nil
	}
	index := len(*s) - 1
	element := (*s)[index]
	*s = (*s)[:index]
	return element
}

func main() {
	var s Stack
	var cmd, v int
	var a bool

	for {
		input := ""
		fmt.Print("Enter command and value: ")
		fmt.Scanln(&input)

		if input == "q" {
			break
		}

		parts := strings.Split(input, " ")
		cmd, _ = strconv.Atoi(parts[0])
		v, _ = strconv.Atoi(parts[1])

		s.Push(((func(x, y int) int { return x }), 0))
		a = true

		if cmd == 1 {
			for !s.IsEmpty() {
				f, _ := s[len(s)-1].(func(int, int) int)
				if f == nil {
					break
				}
				if f(v, 0) <= v {
					s.Pop()
				} else if f == min {
					s = nil
					s.Push((func(x, y int) int { return y }, v))
					a = false
					break
				} else if f == max && f(v, 0) >= v {
					a = false
					break
				}
				break
			}
			if a {
				s.Push(max)
			}
		} else if cmd == 2 {
			for !s.IsEmpty() {
				f, _ := s[len(s)-1].(func(int, int) int)
				if f == nil {
					break
				}
				if f == max && f(v, 0) <= v {
					break
				} else if f == min && f(v, 0) <= v {
					a = false
					break
				} else if f == min && f(v, 0) > v {
					s.Pop()
				} else if f == max && f(v, 0) >= v {
					s = nil
					s.Push((func(x, y int) int { return y }, v))
					a = false
					break
				}
				break
			}
			if a {
				s.Push(min)
			}
		} else if cmd == 3 {
			for _, f := range s {
				v = f.(func(int, int) int)(v, 0)
			}
			fmt.Println(v)
		}
	}
}
